lib/cache_stream.js

```diff
@@ -11,7 +11,7 @@ function CacheStream() {
 require('util').inherits(CacheStream, Transform);
 
 CacheStream.prototype._transform = function(chunk, enc, callback) {
-  var buf = chunk instanceof Buffer ? chunk : Buffer.from(chunk, enc);
+  const buf = chunk instanceof Buffer ? chunk : Buffer.from(chunk, enc);
```

Comment:
I can't get this `Buffer.from` branch to run in the test. Can it ever be something other than Buffer? If not the `Buffer.from` branch can be removed and coverage will be 100%.

Reply:
Or you can do something tricky like this in `./test/cache_stream.js`

```js
it('default', () => {
  const src = new Readable();
  const cacheStream = new CacheStream();
  const content1 = Buffer.from('test1');
  const content2 = 'test2';
  let transformed = false;

  src.push(content1);
  src.push(null);
  src.pipe(cacheStream);

  cacheStream._transform(content2, 'utf8', () => { transformed = true; });

  cacheStream.on('finish', () => {
    cacheStream.getCache().should.eql(content1 + content2);
    transformed.should.be.true;
  });
});
```

Reply:
https://nodejs.org/api/stream.html#stream_transform_transform_chunk_encoding_callback
Even if stream is not an objectMode, it is possible to pass string. For this reason, we need to assume that string will flow.

Reply:
As a result of continuing committing, we found that `chunk` will be string only when `cacheStream.setEncoding('utf8')` is done. In `src.setEncoding('utf 8')`, `chunk` in `_transform()` was converted to `Buffer`.

Reply:
The type of `chank` of `_transform()`
- Set `decodeStrings` of the` Transform (super: Writeble) `class's constructor option to` false`.
- Calling `setEncoding()` on the source `Readable`.
Only when it meets the above conditions, it is `string`, otherwise it is `Buffer`.

Reply:
After finishing the previous behavior check, in the case of the current implementation, we were able to grasp that `chunk` is only `Buffer` ...